riot_pal/dut_pyshell.py

```diff
@@ -0,0 +1,258 @@
+___$$__________$_$$$$_$$__$$$$__$$$____$
+____$$________$$§§§§__$__$$$$$_____$$___$
+_____$$______$$§$§§$_____$$$_______$$$___$
+______$$___$$$§§$§§$$_____________________$
+_______$$_$$$§§$§§$§§$______________$$$$__ $
+_________$$$§§$$§$$§§$_____________$__$$__ $
+_______§$§$§§$$§§$§§§$____________$$$$$___ $
+______§$§$$§$§§§$$§§$_____________________ $
+_____$$§$$$$§§§$$$§$$$$___________________$
+____$$§$$$$§§§$$§$§§$$$$$$______________$$
+____$§§$§$§§§§$§§$§§$$$$___$$$$$$$$$$$$$
+___$$§$§§§§§§$§§§$§§$$$______________$
+___$§$$§§§§§$§§§§$§§$$$$____________$
+____$$§§§§§$§§§§§$§§$$$$____________$
+_____$§§§$$$§§§§§$§§$$$$$$$________$
+____$$§§$§§$§§§§$§§$$$$$$_$________$
+____$§§§§§§$§§§§$§§$$$_____$$_____$
+___$§§§§§§$§§§§$$§$$__________$$_$$
+$$$$§§§§§$$§§§§$§§$_____________$$
+$$§§§§§§$$$$§§$§§$$_______________$
+$§§§§$$§§§§§§$§§$__________________$$
+$§$$§§§§$§§$$§$$____________________$
+$$§$§§§$$§§$§$$$____________________$$
+$$$$§§§§§§§$$$$_____________________$$$
+§$$§§§$§§§§$$$$_____________________$$$
+__$$§§§§$§$§$_______________________$$$
+
+""")
+
+    def do_exit(self, arg):
+        """I mean it should be obvious
+
+        Usage:
+            exit
+        """
+        arg = arg
+        return True
+
+    def _print_func_result_success(self, results):
+        if not isinstance(results, list):
+            results = [results]
+        result = RESULT_SUCCESS
+        printed_something = False
+        for res in results:
+            if self.data_only:
+                if 'result' in res:
+                    if res['result'] == RESULT_SUCCESS:
+                        if 'data' in res:
+                            for data in res['data']:
+                                print(dumps(data))
+                                printed_something = True
+                    else:
+                        result = res['result']
+                else:
+                    print(dumps(res))
+                    printed_something = True
+            else:
+                print(dumps(res))
+                printed_something = True
+        if not printed_something:
+            print(result)
+
+    def _print_func_result(self, func, arg):
+        value = (arg or '').split(' ')
+        func_args = [v for v in value if v]
+        try:
+            results = func(*func_args)
+        except KeyError as exc:
+            print('Could not parse argument {}'.format(exc))
+        except (TypeError, ValueError, SyntaxError) as exc:
+            print(exc)
+        else:
+            self._print_func_result_success(results)
+
+
+PARSER = argparse.ArgumentParser()
+
+LOG_LEVELS = ('debug', 'info', 'warning', 'error', 'fatal', 'critical')
+PARSER.add_argument('--loglevel', choices=LOG_LEVELS, default='info',
+                    help='Python logger log level')
+PARSER.add_argument('--port', '-p', help='Specifies the serial port',
+                    default=None)
+PARSER.add_argument('--data_filter_off', '-do', default=True,
```

Comment:
all short hands short be single letters, and as mentioned above a better name for this would be `debug` hence `-d`

Reply:
I agree names could improve